src/modules/site-v2/base/views/isotype.py

```diff
@@ -47,19 +47,37 @@ def isotype_page(isotype_name, release=None):
     Isotype page
   """
 
+  # Get the list of strains for this isotype, throwing the appropriate error if this fails
+  try:
+    isotype_strains = query_strains(isotype_name=isotype_name)
+  except Exception as ex:
+    logger.error(f'Failed to retrieve strain list for isotype {isotype_name}: {ex}')
+    abort(500)
+  if not isotype_strains:
+    abort(404)
+
+  # Try to sort the list of strains
+  # If this fails for any reason, we can keep going with the unsorted list
+  try:
+    isotype_strains = Strain.sort_by_strain( isotype_strains )
+  except Exception as ex:
+    logger.error(f'Failed to sort strain list for isotype {isotype_name}: {ex}')
+
   try:
-    disable_parent_breadcrumb = True
-    isotype_strains = Strain.sort_by_strain( query_strains(isotype_name=isotype_name) )
     species = isotype_strains[0].species_name
     files = get_blob_list(MODULE_SITE_BUCKET_PHOTOS_NAME, species)
 
     image_urls = {}
     for s in isotype_strains:
       # Get images and thumbs for each strain
       for file in files:
-        start_idx = file.name.index('/')
-        end_idx = file.name.index('.')
-        file_name = file.name[start_idx+1:end_idx]
+        try:
+          start_idx = file.name.index('/')
```

Comment:
@orzut -- instead of parsing this ourselves, can we use pathlib to extract the filename and extension?